lib/icingadb/icingadb.cpp

```diff
@@ -52,9 +57,45 @@ void IcingaDB::Start(bool runtimeCreated)
 {
 	ObjectImpl<IcingaDB>::Start(runtimeCreated);
 
-	boost::call_once([]() {
-		m_EnvironmentId = SHA1(GetEnvironment());
-	}, m_EnvironmentIdOnce);
+	std::call_once(m_EnvironmentIdOnce, []() {
+		String path = Configuration::DataDir + "/icingadb.env";
+
+		if (Utility::PathExists(path)) {
+			std::ifstream fp(path);
+			fp >> m_EnvironmentId;
```

Comment:
You're doing I/O.. like this? >>? Not even the exception flag(s)?

Reply:
While you're on it: consider Boost.

Reply:
Can you be more specific please?

Reply:
Consider looking at Boost, whether it has better I/O mechanisms.

Reply:
My current plan would be to use these functions and just store it as JSON, any objections to that? https://github.com/Icinga/icinga2/blob/f976e351f4047efc24be1cfc78651ab26cb201c1/lib/base/utility.cpp#L1429-L1456

Reply:
Fine.